pkg/controllers/node/termination/terminator/eviction.go

```diff
@@ -197,27 +169,33 @@ func (q *Queue) Evict(ctx context.Context, key QueueKey) bool {
 			// https://github.com/kubernetes/kubernetes/blob/ad19beaa83363de89a7772f4d5af393b85ce5e61/pkg/registry/core/pod/storage/eviction.go#L160
 			// 409 - The pod exists, but it is not the same pod that we initiated the eviction on
 			// https://github.com/kubernetes/kubernetes/blob/ad19beaa83363de89a7772f4d5af393b85ce5e61/pkg/registry/core/pod/storage/eviction.go#L318
-			return true
+			fmt.Printf("Error is an acceptable one\n")
+			return nil
 		}
 		if apierrors.IsTooManyRequests(err) { // 429 - PDB violation
 			q.recorder.Publish(terminatorevents.NodeFailedToDrain(&corev1.Node{ObjectMeta: metav1.ObjectMeta{
-				Name:      key.Name,
-				Namespace: key.Namespace,
-			}}, serrors.Wrap(fmt.Errorf("evicting pod violates a PDB"), "Pod", klog.KRef(key.Namespace, key.Name))))
-			return false
+				Name:      pod.Name,
```

Comment:
This event looks wrong -- it's referencing a node but that node is based on the pod info?

Reply:
Yeah this event seems to be bugged in the current release of Karpenter. Here's an example of this event from our datadog event collector:

![image](https://github.com/user-attachments/assets/c94a913a-b41d-45c8-8da8-473ddff4d0ff)

We have no nodes named "arc/arc-4", that's a namespace and statefulset pod name.